lablup/backend.ai
# impl feature to force password updates periodically

related issue: https://github.com/lablup/giftbox/issues/343

impl feature to force password updates periodically.

---
## How to set up for test

1. upgrade migration by running `./py -m alembic upgrade head`
2. insert auth config to etcd
ex - save the json below to a <FILE_NAME> and put it by running
`backend.ai mgr etcd put-json config/auth <FILE_NAME>`
```
{
  "max_password_age": "1m",
  "shared_secret": "random",
  "exp": "5m"
}
```
"1m" means 1 minute! Available units to represent the time above are "w, d, h, m, s".

3. create user and check it!

---

## APIs
```
# Direct to manager
POST <MANAGER_URL>/auth/update-password-no-auth
{
  "domain" : "",
  "username" : "",
  "current_password": "",
  "new_password": ""
}

# To webserver
POST <WEBSERVER_URL>/server/update-password-no-auth
{
  "username" : "",
  "current_password": "",
  "new_password": ""
}

```

## Example request

<img width="862" alt="image" src="https://github.com/lablup/backend.ai/assets/44239739/37f6630a-e25e-46d7-808d-0fb0f291fa45">

```
curl --request POST \
  --url http://127.0.0.1:8090/server/update-password-no-auth \
  --header 'Content-Type: application/json' \
  --header 'Date: 2023-05-23T08:59:49.685263+00:00' \
  --header 'X-BackendAI-Domain: default' \
  --header 'X-BackendAI-Version: v6.20230315' \
  --data '{
	"username": "test@email.com",
	"current_password": "1234",
	"new_password": "1234"
}
'
```
